### PR TITLE
portsyncd: Adding bring up port automatically logic

### PR DESCRIPTION
**Pull request by stcheng**

### CODE REVIEW

File: portsyncd/linksync.cpp [hunk 1]
```diff
@@ -44,19 +42,45 @@ void LinkSync::onMsg(int nlmsg_type, struct nl_object *obj)
...
+                init = false;
+                break;
+            }
+        }
+
+        if (init)
+        {
+            int rv = system(("/sbin/ifup --force " + key).c_str());
+            if (!rv)
+                cout << "failed to bring up port " + key << endl;
```
Comment (kcudnik) [inline portsyncd/linksync.cpp]: swss_log_err ?

File: portsyncd/linksync.cpp [hunk 2]
```diff
@@ -44,19 +42,45 @@ void LinkSync::onMsg(int nlmsg_type, struct nl_object *obj)
...
     if (m_lagTableConsumer.get(key, temp))
         m_lagTableProducer.set(key, fvVector);
     else if (m_vlanTableConsumer.get(key, temp))
         m_vlanTableProducer.set(key, fvVector);
     else if (m_portTableConsumer.get(key, temp))
-        m_portTableProducer.set(key, fvVector);
+    {
+        /*
+         * If the port entry is just initialized without having admin_status/oper_status field,
+         * we manually bring up the port using ifup --force command.
```
Comment (lguohan) [inline portsyncd/linksync.cpp]: this port bring up logic should be the same for lag, port and vlan, why just limit to port?

File: portsyncd/linksync.cpp [hunk 3]
```diff
@@ -44,19 +42,45 @@ void LinkSync::onMsg(int nlmsg_type, struct nl_object *obj)
...
+         */
+        bool init = true;
+        for (auto it = temp.begin(); it != temp.end(); it++)
+        {
+            if (fvField(*it) == "admin_status" || fvField(*it) == "oper_status")
```
Comment (lguohan) [inline portsyncd/linksync.cpp]: why check oper_status? the port orch may set the oper_status, but the linux created the host interface.

File: portsyncd/linksync.cpp (hunk 2 above)
Comment (lguohan) [inline portsyncd/linksync.cpp]: "If the port entry is just initialized without having admin_status/oper_status field", i do not know if this is good assuming or not.

probably the correct solution should read the /etc/network/interface file and find out whether the port should be up or down.

Comment (stcheng) [inline portsyncd/linksync.cpp]: vlan and lag support will be introduced later.

**Comment by lguohan** [comment]
ack

**Pull request merged**